pisek/task_jobs/program.py

```diff
@@ -91,12 +91,17 @@ def __init__(self, env: Env, name: str, **kwargs) -> None:
     def _load_compiled(self, program: TaskPath) -> TaskPath:
         """Loads name of compiled program."""
         executable = TaskPath.executable_file(self._env, program.path)
-        if not self._file_exists(executable):
+        executable_dir = TaskPath.executable_file(
+            self._env, os.path.join(program.path, "run")
+        )
+
+        if self._is_file(executable) or self._is_file(executable_dir):
+            return executable
+        else:
```

Comment:
I don't think it makes much sense to construct the executable path twice, once in `_load_compiled` and once in `_load_executable`. Can't they be combined somehow, or could `_load_compiled` return the path and mark the files as accessed?